packages/components/src/components/form/SelectBar/SelectBar.tsx

```diff
@@ -59,10 +44,63 @@ const getTranslateValue = (index: number) => {
 const getPuckWidth = (optionsCount: number) =>
     `calc((100% - 8px - ${(optionsCount - 1) * spacings.xxs}px) / ${optionsCount})`;
 
+const columnWrapper = css`
+    flex-direction: column;
+    align-items: flex-start;
+    width: 100%;
+`;
+const columnOption = css`
+    grid-auto-flow: row;
+    width: 100%;
+    border-radius: ${borders.radii.lg};
+`;
+const columnPuck = ({
+    $selectedIndex: selectedIndex,
+    $optionsCount,
+}: {
+    $optionsCount: number;
+    $selectedIndex: number;
+}) => css`
+    left: 4px;
+    right: 4px;
+    top: 4px;
+    width: auto;
+    height: ${getPuckWidth($optionsCount)};
+    transform: ${`translateY(${getTranslateValue(selectedIndex)})`};
+`;
+
+const Wrapper = styled.div<
+    TransientProps<AllowedFrameProps> & { $isFullWidth?: boolean; $orientation: Orientation }
+>`
+    display: flex;
+    align-items: center;
+    gap: ${spacingsPx.sm};
+    width: ${({ $isFullWidth }) => ($isFullWidth ? '100%' : 'auto')};
+
+    ${({ $orientation }) =>
+        $orientation === 'auto' &&
+        css`
+            ${breakpointMediaQueries.below_sm} {
+                ${columnWrapper}
+            }
+        `}
+
+    ${({ $orientation }) => $orientation === 'vertical' && columnWrapper}
```

Comment:
Not 100% happy with this, if you have better idea how to do it, feel free to comment it